### PR TITLE
Milestones confirmed

### LINKED ISSUE #171: Client - MQTT milestones/confirmed subscription
## Description

Implements the `milestones/confirmed` event API.

### PR DESCRIPTION
**Pull request by rebinmannady93**
# Description of change

fixes #171

## Type of change

Choose a type of change, and delete any options that are not relevant.

- Enhancement (a non-breaking change which adds functionality)

## How the change has been tested

Test case : test_milestones_confirmed_event

## Change checklist

Add an `x` to the boxes that are relevant to your changes, and delete any items that are not.

- [x] My code follows the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [] I have made corresponding changes to the documentation
- [x] I have added tests using CTest that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: tests/client/api_events/test_milestones_confirmed_event.c [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+      TEST_ASSERT_EQUAL_INT(0, event_stop(event->client));
+      break;
+    default:
+      break;
+  }
+}
+
+void test_milestones_confirmed_events(void) {
+  event_client_config_t config = {
+      .host = "mqtt.lb-0.h.chrysalis-devnet.iota.cafe", .port = 1883, .client_id = "iota_test_1", .keepalive = 60};
```
Comment (oopsmonk) [inline tests/client/api_events/test_milestones_confirmed_event.c]: please use predefined configurations, for example `TEST_EVENTS_HOST`, in the event_test_config.h, or remove event_test_config.h if not used? 
same as `test_milestones_latest_event.c` test case.

File: tests/client/api_events/test_milestones_confirmed_event.c [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+void setUp(void) {}
+
+void tearDown(void) {}
+
+void test_milestones_confirmed_parser(void) {
+  char *json_data = "{\"index\":1139767,\"timestamp\": 1637221249}";
+
+  // Test for expected events response
+  milestone_confirmed_t *res = res_milestones_confirmed_new();
```
Comment (oopsmonk) [inline tests/client/api_events/test_milestones_confirmed_event.c]: don't think dynamic allocation is needed for this response object. 
the user can simply use it like
```
milestone_confirmed_t res = {};
parse_milestones_confirmed(json_data, &res)
```
so `res_milestones_confirmed_new` and `res_milestones_confirmed_free` are useless.

File: src/client/api/events/sub_milestones_confirmed.h [hunk 3]
```diff
@@ -0,0 +1,47 @@
+// Copyright 2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef __SUB_MILESTONES_CONFIRMED_H__
+#define __SUB_MILESTONES_CONFIRMED_H__
+
+#include <stdbool.h>
```
Comment (oopsmonk) [inline src/client/api/events/sub_milestones_confirmed.h]: unused header file.

**Review by oopsmonk — CHANGES_REQUESTED** [review]

File: tests/client/api_events/test_milestones_confirmed_event.c (hunk 1 above)
Comment (rebinmannady93) [reply tests/client/api_events/test_milestones_confirmed_event.c]: Done.

File: tests/client/api_events/test_milestones_confirmed_event.c (hunk 2 above)
Comment (rebinmannady93) [reply tests/client/api_events/test_milestones_confirmed_event.c]: Removed dynamic allocation for milestones_confirmed and milestones_lates topic.

File: src/client/api/events/sub_milestones_confirmed.h (hunk 3 above)
Comment (rebinmannady93) [reply src/client/api/events/sub_milestones_confirmed.h]: Removed.

**Review by oopsmonk — APPROVED** [review]

**Pull request merged**